python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py

```diff
@@ -167,12 +167,11 @@ async def handle_start(self, message: GroupChatStart, ctx: MessageContext) -> No
 
     @event
     async def handle_agent_response(self, message: GroupChatAgentResponse, ctx: MessageContext) -> None:  # type: ignore
-        self._message_thread.append(message.agent_response.chat_message)
         delta: List[AgentMessage] = []
         if message.agent_response.inner_messages is not None:
             for inner_message in message.agent_response.inner_messages:
-                self._message_thread.append(inner_message)
                 delta.append(inner_message)
+        self._message_thread.append(message.agent_response.chat_message)
```

Comment:
Side comment?
What is the main reason for not appending inner_messages? 
(e.g., our current convention that inner messages are not per se useful to task execution but more as event updates for applications?)

Reply:
By design, the M1 orchestrator does not care about the inner messages. Each agent is supposed to return the relevant info in the final response they send.

Reply:
We could label each message as inner message or not and handle this. again future PR